presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDifferenceFunction.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PageBuilder;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import it.unimi.dsi.fastutil.ints.AbstractIntComparator;
+import it.unimi.dsi.fastutil.ints.IntArrays;
+import it.unimi.dsi.fastutil.ints.IntComparator;
+
+@Description("Set difference of two arrays")
+@ScalarFunction("array_diff")
+public final class ArrayDifferenceFunction
+{
+    private static final int INITIAL_SIZE = 128;
+
+    private int[] leftPositions = new int[INITIAL_SIZE];
+    private int[] rightPositions = new int[INITIAL_SIZE];
+    private final PageBuilder pageBuilder;
+
+    @TypeParameter("E")
+    public ArrayDifferenceFunction(@TypeParameter("E") Type elementType)
+    {
+        pageBuilder = new PageBuilder(ImmutableList.of(elementType));
+    }
+
+    private static IntComparator IntBlockCompare(Type type, Block block)
+    {
+        return new AbstractIntComparator()
+        {
+            @Override
+            public int compare(int left, int right)
+            {
+                if (block.isNull(left) && block.isNull(right)) {
+                    return 0;
+                }
+                if (block.isNull(left)) {
+                    return -1;
+                }
+                if (block.isNull(right)) {
+                    return 1;
+                }
+                return type.compareTo(block, left, block, right);
+            }
+        };
+    }
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public Block difference(
+            @TypeParameter("E") Type type,
+            @SqlType("array(E)") Block left,
```

Comment:
While I do not feel particularly strongly about this, I would prefer that these two variables be called `leftArray` and `rightArray`.